pkg/agent/discoverycache.go

```diff
@@ -125,10 +127,20 @@ func (j *discoveryCache) updateAPICache() error {
 		return err
 	}
 
+	// Check if marketplace provisioning is enabled.  If it is enabled, verify that current apiservices have the necessary request definitions
+	if agent.agentFeaturesCfg.MarketplaceProvisioningEnabled() {
+		j.logger.Trace("marketplace provisioning migration starting")
+		j.marketplaceProvisioningMigration(apiServices)
+	}
+
+	return j.attributeMigration(apiServices, existingAPIs)
+}
+
+func (j *discoveryCache) attributeMigration(apiServices []*apiV1.ResourceInstance, existingAPIs map[string]bool) error {
```

Comment:
Maybe we can have an array of migrate types that implement a `Migration` interface that we can loop through and pass each service too so we can keep the specifics of what is being migrated out of the discovery cache

Reply:
Or we can add another type that implements the `Migration` interface that contains a list of all the migrations to run, and then as we loop through each service we just pass the service into `Migrate` one time, and that type will handle all the other migrations.